### DISCUSSION TITLE
ADAM optimizer support for training

### DISCUSSION THREAD

**Discussion by ATirelli** [deepmodeling/deepmd-kit#307] [Q&A] [answered]
Hello! I was wondering whether you it would be possible to integrate the ADAM optimiser as an option for the training process. By looking at the code, it looks like the only learning rate schedule available at the moment is Exponential Decay. 

Thanks in advance!

**Comment by njzjz** [comment, marked as answer]
https://github.com/deepmodeling/deepmd-kit/blob/b2662e2516fa6decef20a2fa183a989d2394e3c7/source/train/Trainer.py#L278
The default optimizer is Adam. Exponential Decay is used for the learning rate of Adam. You can see tensorflow API docs https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adam.

**Comment by ATirelli** [comment]
Oh, I see. Thank you!
